tests/components/anglian_water/test_config_flow.py

```diff
@@ -0,0 +1,68 @@
+"""Test Anova config flow."""
+
+from unittest.mock import patch
+
+from pyanglianwater import API
+from pyanglianwater.exceptions import InvalidUsernameError
+
+from homeassistant import config_entries
+from homeassistant.components.anglian_water.const import DOMAIN
+from homeassistant.const import CONF_DEVICES, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from . import CONF_INPUT
+
+
+async def test_flow_user(hass: HomeAssistant, anglian_api: API) -> None:
+    """Test user initialized flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_USER},
+    )
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input=CONF_INPUT,
+    )
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["data"] == {
+        CONF_USERNAME: "sample@gmail.com",
+        CONF_PASSWORD: "sample",
+        CONF_DEVICES: [],
+    }
+
+
+async def test_flow_wrong_login(hass: HomeAssistant) -> None:
+    """Test incorrect login throwing error."""
+    with patch(
+        "homeassistant.components.anglian_water.config_flow.API.create_via_login",
+        side_effect=InvalidUsernameError,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+        )
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input=CONF_INPUT,
+        )
+        assert result["type"] is FlowResultType.FORM
+        assert result["errors"] == {"base": "invalid_auth"}
+
+
+async def test_flow_unknown_error(hass: HomeAssistant) -> None:
+    """Test unknown error throwing error."""
+    with patch(
+        "homeassistant.components.anova.config_flow.API.create_via_login",
+        side_effect=Exception(),
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+        )
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input=CONF_INPUT,
+        )
+        assert result["type"] is FlowResultType.FORM
+        assert result["errors"] == {"base": "unknown"}
```

Comment:
Both tests need to end in a CREATE_ENTRY to completely test that we are able to recover from an error. You can use `@pytest.mark.parametrize` for this